rpcs3/Emu/CPU/CPUThread.cpp

```diff
@@ -1131,19 +1111,20 @@ void cpu_thread::stop_all() noexcept
 	{
 		auto on_stop = [](u32, cpu_thread& cpu)
 		{
-			cpu.abort();
+			cpu.state += cpu_flag::exit;
+			cpu.state.notify_one(cpu_flag::exit);
 		};
 
 		idm::select<named_thread<ppu_thread>>(on_stop);
 		idm::select<named_thread<spu_thread>>(on_stop);
 	}
 
 	sys_log.notice("All CPU threads have been signaled.");
+}
 
-	while (s_cpu_counter)
-	{
-		std::this_thread::sleep_for(1ms);
-	}
+void cpu_thread::cleanup() noexcept
+{
+	ensure(!s_cpu_counter);
```

Comment:
since s_cpu_counter is decreased in a seperate thread than the main thread, we cannot be sure that s_spu_counter is already 0 when we call cleanup after stopping all threads in System.cpp.
Replacing the ensure with `while (s_cpu_counter) std::this_thread::sleep_for(1ms);` fixes the crashes for me.
(I'm not claiming that this is the ideal solution)